src/Server/src/GraphQL/Client/TokenRequestDataResultFactory.cs

```diff
@@ -36,28 +38,32 @@ public class TokenRequestDataResultFactory : IResultFactory<TokenRequestData, To
 
         var clientId = client.ClientId;
 
-        var secretEncrypted =
-            client.ClientSecrets.First(secret => secret.Id.Equals(input.SecretId)).EncryptedValue;
+        var secretEncrypted = client.ClientSecrets.First(secret => secret.Id.Equals(input.SecretId))
+            .EncryptedValue;
+        if (secretEncrypted == null)
+        {
+            throw new KeyNotFoundException("No encrypted secret found");
+        }
+
         var secretDecrypted =
             await _encryptionService.DecryptAsync(secretEncrypted, cancellationToken);
 
-        var grantTypes = client.AllowedGrantTypes.First();
+        if (client.AllowedGrantTypes == null)
+        {
+            throw new ArgumentNullException("No grant types found");
+        }
+
+        string grantTypes = string.Join(", ", client.AllowedGrantTypes);
```

Comment:
@fnymmm Are you sure this works? We cannot request a token for more than one grant type so we will probably run into an issue if we have a client with multiple grant types.